src/cloudsc_python/src/cloudsc4py/physics/_stencils/cloudsc.py

```diff
@@ -0,0 +1,2176 @@
+        tmp_trpaus[0, 0] = 0.1
+        tmp_paphd[0, 0] = 1 / tmp_aph_s[0, 0]
+    with computation(FORWARD), interval(0, -2):
+        sig = in_ap[0, 0, 0] * tmp_paphd[0, 0]
+        if sig > 0.1 and sig < 0.4 and t[0, 0, 0] > t[0, 0, 1]:
+            tmp_trpaus[0, 0] = sig
+
+    # === 3: physics
+    # --- main vertical loop
+    with computation(FORWARD):
+        with interval(0, NCLDTOP - 1):
+            # --- initialize variables
+            lude = in_lude[0, 0, 0]
+            pfplsl = 0.0
+            pfplsi = 0.0
+            pfplsr = 0.0
+            pfplss = 0.0
+            pfplsv = 0.0
+            qln = 0.0
+            qin = 0.0
+            qrn = 0.0
+            qsn = 0.0
+            qvn = 0.0
+            anew = 0.0
+        with interval(NCLDTOP - 1, -1):
+            # *** 3.0: initialize variables
+            # --- first guess microphysics
+            qlfg = ql
+            qifg = qi
+            qrfg = qr
+            qsfg = qs
+            qvfg = qv
+
+            convsink_ql = 0.0
+            convsink_qi = 0.0
+            convsink_qr = 0.0
+            convsink_qs = 0.0
+            convsrce_ql = 0.0
+            convsrce_qi = 0.0
+            convsrce_qr = 0.0
+            convsrce_qs = 0.0
+            convsrce_qv = 0.0
+            fallsrce_ql = 0.0
+            fallsrce_qi = 0.0
+            fallsrce_qr = 0.0
+            fallsrce_qs = 0.0
+            index1_ql = True
+            index1_qi = True
+            index1_qr = True
+            index1_qs = True
+            index1_qv = True
+            index3_ql_ql = False
+            index3_ql_qi = False
+            index3_ql_qr = False
+            index3_ql_qs = False
+            index3_ql_qv = False
+            index3_qi_ql = False
+            index3_qi_qi = False
+            index3_qi_qr = False
+            index3_qi_qs = False
+            index3_qi_qv = False
+            index3_qr_ql = False
+            index3_qr_qi = False
+            index3_qr_qr = False
+            index3_qr_qs = False
+            index3_qr_qv = False
+            index3_qs_ql = False
+            index3_qs_qi = False
+            index3_qs_qr = False
+            index3_qs_qs = False
+            index3_qs_qv = False
+            index3_qv_ql = False
+            index3_qv_qi = False
+            index3_qv_qr = False
+            index3_qv_qs = False
+            index3_qv_qv = False
+            lcust_ql = 0.0
+            lcust_qi = 0.0
+            lcust_qr = 0.0
+            lcust_qs = 0.0
+            lcust_qv = 0.0
+            ldefr = 0.0
+            lfinalsum = 0.0
+            order_ql = -999
+            order_qi = -999
+            order_qr = -999
+            order_qs = -999
+            order_qv = -999
+            psupsatsrce_ql = 0.0
+            psupsatsrce_qi = 0.0
+            psupsatsrce_qr = 0.0
+            psupsatsrce_qs = 0.0
+            qpretot = 0.0
+            solab = 0.0
+            solac = 0.0
+            solqa_ql_ql = 0.0
```

Comment:
This is ok as is, but just for my own education, would it be possible to add a a custom constant dimension like `NCLV` be possible at some point to avoid this unrolling?

Reply:
As we discussed offline, GT4Py doesn't support for-loops over the data dimensions (only vectorized operations, which are not really useful for us). I totally agree that this feature would make the code way more pretty. I will ask the GT4Py dev team whether there exists any plan to add support for this.